src/main/java/enums/Currency.java

```diff
@@ -8,9 +8,18 @@
 @ToString
 @RequiredArgsConstructor
 public enum Currency {
-    EUR("€"),
-    GBP("£"),
-    USD("$");
+	EUR("€"),
+	GBP("£"),
+	USD("$");
 
-    public final String currencySymbol;
+	public final String currencySymbol;
+
+	public static Currency fromSymbol(String symbol) {
+		for (Currency currency : values()) {
+			if (currency.getCurrencySymbol().equals(symbol)) {
```

Comment:
You can use stream() here as well :)

Reply:
It was not easy and, in addition, the solution seems less readable. What do you think?
```java
public static Currency fromSymbol(String symbol) {
    return Arrays.stream(values())
        .filter(currency -> currency.getCurrencySymbol().equals(symbol))
        .findFirst()
        .orElseThrow(() -> new IllegalArgumentException("No enum constant for symbol: " + symbol));
}
```